File: src/main/java/com/cuupa/dms/authentication/CurrentUser.java
package com.cuupa.dms.authentication;

import com.vaadin.flow.server.VaadinRequest;
import com.vaadin.flow.server.VaadinService;

import java.util.Objects;

public final class CurrentUser {

    /**
     * The attribute key used to store the username in the session.
     */
    public static final String CURRENT_USER_SESSION_ATTRIBUTE_KEY = CurrentUser.class.getCanonicalName();

    private CurrentUser() {
    }

    /**
     * Returns the name of the current user stored in the current session, or an
     * empty string if no user name is stored.
     *
     * @throws IllegalStateException if the current session cannot be accessed.
     */
    public static User get() {
        User
                currentUser =
                (User) getCurrentRequest().getWrappedSession().getAttribute(CURRENT_USER_SESSION_ATTRIBUTE_KEY);
        return Objects.requireNonNullElse(currentUser, new User());
    }

    /**
     * Sets the name of the current user and stores it in the current session.
     * Using a {@code null} username will remove the username from the session.
     *
     * @throws IllegalStateException if the current session cannot be accessed.
     */
    public static void set(User currentUser) {
        if (currentUser == null) {
            getCurrentRequest().getWrappedSession().removeAttribute(CURRENT_USER_SESSION_ATTRIBUTE_KEY);
        } else {
            getCurrentRequest().getWrappedSession().setAttribute(CURRENT_USER_SESSION_ATTRIBUTE_KEY, currentUser);
        }
    }

    private static VaadinRequest getCurrentRequest() {
        VaadinRequest request = VaadinService.getCurrentRequest();
        if (request == null) {
            throw new IllegalStateException("No request bound to current thread.");
        }
        return request;
    }
}

File: src/main/java/com/cuupa/dms/authentication/CurrentUser.kt
package com.cuupa.dms.authentication

import com.vaadin.flow.server.VaadinRequest
import com.vaadin.flow.server.VaadinService

object CurrentUser {
    /**
     * The attribute key used to store the username in the session.
     */
    private val CURRENT_USER_SESSION_ATTRIBUTE_KEY: String = CurrentUser::class.java.canonicalName

    /**
     * Returns the name of the current user stored in the current session, or an
     * empty string if no user name is stored.
     *
     * @throws IllegalStateException if the current session cannot be accessed.
     */
    @JvmStatic
    fun get(): User {
        val currentUser: User? = currentRequest.wrappedSession.getAttribute(CURRENT_USER_SESSION_ATTRIBUTE_KEY) as User?
        return currentUser ?: User()
    }

    /**
     * Sets the name of the current user and stores it in the current session.
     * Using a `null` username will remove the username from the session.
     *
     * @throws IllegalStateException if the current session cannot be accessed.
     */
    @JvmStatic
    fun set(currentUser: User?) {
        if (currentUser == null) {
            currentRequest.wrappedSession.removeAttribute(CURRENT_USER_SESSION_ATTRIBUTE_KEY)
        } else {
            currentRequest.wrappedSession.setAttribute(CURRENT_USER_SESSION_ATTRIBUTE_KEY, currentUser)
        }
    }

    private val currentRequest: VaadinRequest
        get() = VaadinService.getCurrentRequest()
                ?: throw IllegalStateException("No request bound to current thread.")
}